playground/backend/internal/setup_tools/builder/setup_builder_test.go

```diff
@@ -23,100 +23,293 @@ import (
+			name: "Test correct compiler builder",
+			args: args{
+				lc:     lc,
+				sdkEnv: sdkEnv,
+			},
+			want: &wantExecutor.ExecutorBuilder,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := SetupCompilerBuilder(tt.args.lc, tt.args.sdkEnv)
+			if !reflect.DeepEqual(fmt.Sprint(got.Build()), fmt.Sprint(tt.want.Build())) {
+				t.Errorf("SetupCompilerBuilder() = %v, want %v", got.Build(), tt.want.Build())
+			}
+		})
+	}
+}
+
+func TestSetupRunBuilder(t *testing.T) {
+	wantExecutor := executors.NewExecutorBuilder().
+		WithRunner().
+		WithExecutableFileName(lc.GetAbsoluteExecutableFilePath()).
+		WithWorkingDir(lc.GetAbsoluteBaseFolderPath()).
+		WithCommand(sdkEnv.ExecutorConfig.RunCmd).
+		WithArgs(sdkEnv.ExecutorConfig.RunArgs).
+		WithPipelineOptions(strings.Split("", " "))
+
+	type args struct {
+		lc              *fs_tool.LifeCycle
+		pipelineOptions string
+		sdkEnv          *environment.BeamEnvs
+	}
+	tests := []struct {
+		name string
+		args args
+		want *executors.ExecutorBuilder
+	}{
+		{
+			// Test case with calling Setup with correct data.
+			// As a result, want to receive an expected run builder.
+			name: "Test correct run builder",
+			args: args{
+				lc:     lc,
+				sdkEnv: sdkEnv,
+			},
+			want: &wantExecutor.ExecutorBuilder,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, _ := SetupRunBuilder(tt.args.lc, tt.args.pipelineOptions, tt.args.sdkEnv)
+			if !reflect.DeepEqual(fmt.Sprint(got.Build()), fmt.Sprint(tt.want.Build())) {
+				t.Errorf("SetupRunBuilder() got = %v, want %v", got.Build(), tt.want.Build())
+			}
+		})
+	}
+}
+
+func TestSetupTestBuilder(t *testing.T) {
+	wantExecutor := executors.NewExecutorBuilder().
+		WithTestRunner().
+		WithExecutableFileName(lc.GetAbsoluteExecutableFilePath()).
+		WithCommand(sdkEnv.ExecutorConfig.TestCmd).
+		WithArgs(sdkEnv.ExecutorConfig.TestArgs).
+		WithWorkingDir(lc.GetAbsoluteSourceFolderPath())
+
+	type args struct {
+		lc     *fs_tool.LifeCycle
+		sdkEnv *environment.BeamEnvs
+	}
+	tests := []struct {
+		name string
+		args args
+		want *executors.ExecutorBuilder
+	}{
+		{
+			// Test case with calling Setup with correct data.
+			// As a result, want to receive an expected test builder.
+			name: "Test correct test builder",
+			args: args{
+				lc:     lc,
+				sdkEnv: sdkEnv,
+			},
+			want: &wantExecutor.ExecutorBuilder,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, _ := SetupTestBuilder(tt.args.lc, tt.args.sdkEnv)
+			if !reflect.DeepEqual(fmt.Sprint(got.Build()), fmt.Sprint(tt.want.Build())) {
+				t.Errorf("SetupTestBuilder() got = %v, want %v", got.Build(), tt.want.Build())
+			}
+		})
+	}
+}
+
+func Test_replaceLogPlaceholder(t *testing.T) {
+	type args struct {
+		lc             *fs_tool.LifeCycle
+		executorConfig *environment.ExecutorConfig
+	}
+	tests := []struct {
+		name string
+		args args
+		want []string
+	}{
+		{
+			name: "",
```

Comment:
Could you please give a name for this test?

Reply:
I will :)

Reply:
Done